script.js

```diff
@@ -341,7 +350,16 @@ function addUpStudentCredits(reportCardTableElement) {
  */
 
 function calculateSemesterGpa(reportCardTableElement) {
-  // code goes here
+  let gpaPointsSum = 0;
+  let gpaElems = reportCardTableElement.querySelectorAll('.gpa');
+  for (let elem of gpaElems) {
+    if (elem.innerHTML != "") {
+      gpaPointsSum += gpaPointsLookup[elem.innerHTML];
```

Comment:
What happens if `gpaPointsLookup` is null? Think about how you would handle that case.